examples/python/pre-filter/pre-filter/handler.py

```diff
@@ -190,37 +330,56 @@ def handle(req):
-    for name, value in event.items():
-        if type(value) == dict:  # a dict is probably a VC managed object reference
-            for moName, moRef in value.items():
-                # Search the items of the dict and look for contained dicts with a 'Type' property
-                if type(moRef) == dict and 'Type' in moRef:
-                    # Get the relevant filter_... environment variable
-                    objFilter = eval(f'os.getenv("filter_{name.lower()}", default=".*")')
-                    # Get a vc managed object
-                    mo = getManagedObject(moRef)
-                    # Test the filter
-                    moType = getManagedObjectTypeName(mo)
-                    debug(f'{bgc.OKBLUE}Apply Filter > {bgc.ENDC}"{name.lower()}" object ({moType}): "filter_{name.lower()}" = {objFilter}')
-                    res = filterVcObject(mo, objFilter)
-                    if isinstance(res, tuple):  # Tuple is returned if the object didn't match the filter
-                        return res
-
-    func = os.getenv("call_function", False)
-    if func:
+
+    # Load filter_... environment variables and strip prefix
+    filters = {}
+    for k, v in os.environ.items():
+        if k.startswith("filter_"):
+            filters[k[7:]] = v
+
+    # Log all available event keys
+    if DEBUG:
+        for key in getEventKeys(event):
+            debug(f"Event Data > {key}")
+
+    # Loop through each defined filter and apply it if possible
+    for key, filter in filters.copy().items():
+        debug(f'Key > {key}')
+        res = filterItem(key, filter, event)
+        if isinstance(res, tuple):  # Tuple is returned if the object didn't match the filter
+            return res
+        if res is True:
+            filters.pop(key)
+
+    # Test if all filters have passed or not
+    if len(filters) > 0 and MATCH_ALL is True:
+        debug(f'{bgc.WARNING}Some filters not matched when match_all is true - "{", ".join(filters.keys())}". Exiting{bgc.ENDC}')
+        return 'some filters not matched', 400
+
+    # All filters matched succesfully so lets call the chained function
+    func = os.getenv("call_function", default="")
+    if func is not "":
         debug(f'{bgc.HEADER}All filters matched. Calling chained function {func}{bgc.ENDC}')
 
+        # Add this functions name to a faasstack array in the cloud event data
+        # This allows the chained function to know who called it
+        if 'faasstack' in cevent:
+            cevent['faasstack'].append(faasFunctionName())
+        else:
+            cevent['faasstack'] = [faasFunctionName()]
+
+        # Send a post request to the chained function with the original event data
         try:
-            res = requests.post(f'http://gateway.openfaas:8080/function/{func}', req)
+            res = requests.post(f'http://gateway.openfaas:8080/function/{func}', json.dumps(cevent))
             return res.text, res.status_code
```

Comment:
This works nicely with the retries we're adding in #241 so you get automatic retries for free here.